### PR TITLE
Fix bug where grapple hook sticks to walls and ground

### PR DESCRIPTION
**Pull request by clayton-snyder**
- Firing at or past a wall or the ground will now result in the grappling hook colliding with the surface and then resetting, as though it had reached its endpoint
- Collision isn't ignored, but casting is set to false causing the hook to reset. 

**5 min**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -418,3 +418,4 @@ ASALocalRun/
 .mfractor/
 
 
+385/.vscode/settings.json
```
Comment (Chris-Johnston) [inline .gitignore]: Should actually use this gitignore instead.

https://github.com/github/gitignore/blob/master/Global/VisualStudioCode.gitignore

File: 385/Assets/Scripts/FireGrappleHook.cs [hunk 2]
```diff
@@ -134,11 +134,16 @@ private void OnTriggerEnter2D(Collider2D collider)
...
+		string colliderTag = collision.collider.tag;
 		// Don't check for collisions with the Player object
 		if (collision.collider.tag == "Player") 
 		{
-			Physics2D.IgnoreCollision (gameObject.GetComponent<Collider2D>(), collision.collider);
-			return;
+			Physics2D.IgnoreCollision (gameObject.GetComponent<Collider2D> (), collision.collider);
+		} 
+		// If the hook collides with the floor or a wall, reset
+		else if (colliderTag == "Ground" || colliderTag == "Floor & Wall")
```
Comment (Chris-Johnston) [inline 385/Assets/Scripts/FireGrappleHook.cs]: can we define these as constant strings?

File: 385/Assets/Scripts/FireGrappleHook.cs [hunk 3]
```diff
@@ -134,11 +134,16 @@ private void OnTriggerEnter2D(Collider2D collider)
     // be initiated).
     void OnCollisionEnter2D(Collision2D collision)
 	{
+		string colliderTag = collision.collider.tag;
 		// Don't check for collisions with the Player object
 		if (collision.collider.tag == "Player")
```
Comment (Chris-Johnston) [inline 385/Assets/Scripts/FireGrappleHook.cs]: should probably use the `colliderTag` here then

File: 385/Assets/Scripts/FireGrappleHook.cs (hunk 2 above)
Comment (clayton-snyder) [reply 385/Assets/Scripts/FireGrappleHook.cs]: good idea. I'll format the tag constant variables as TAG_NAME_HERE if that's fine

Comment (Chris-Johnston) [reply 385/Assets/Scripts/FireGrappleHook.cs]: see PlayerController as a reference

**Pull request merged**